controllers/flinkcluster_converter.go

```diff
@@ -740,11 +730,27 @@ func getDesiredJob(
 	return job
 }
 
-func convertFromSavepoint(
-	jobSpec *v1beta1.JobSpec, jobStatus *v1beta1.JobStatus) *string {
+func convertFromSavepoint(jobSpec *v1beta1.JobSpec, clusterStatus *v1beta1.FlinkClusterStatus) *string {
+	var jobStatus = clusterStatus.Components.Job
+	// 1) When operator should restart the  job for some reason, we want the job resume at the stopped point,
+	// so the operator will use the latest savepoint recorded in savepointLocation in job status.
+	//
+	// 2) In the case of job update
+	// 2-1) User may want to replay the job from the savepoint user provided by updating spec.fromSavepoint.
+	// 2-2) Or user may want to resume the job from the latest savepoint after job update just like when job restarting.
+	// In this case, user must make spec.fromSavepoint empty so that the operator know the job should resume from the latest savepoint.
```

Comment:
Then we might want to set the default value of spec.job.fromSavepoint to `LATEST`, right? Otherwise, since it is optional, if the value is not set initially, if the user updates it without changing it to `LATEST`, the job will start from the beginning, so it may be different than expected.

Reply:
Currently fixed this as your suggestion.

Reply:
> Then we might want to set the default value of `spec.job.fromSavepoint` to `LATEST`, right? Otherwise, since it is optional, if the value is not set initially, when the user updates FlinkCluster without changing it to `LATEST`, the job will start from the beginning. In my opinion, it seems natural to resume the job from the latest savepoint for update as the default behavior, what do you think?

@functicons What do you think about this?